src/components/DialogBox/HexBinDec.vue

```diff
@@ -1,136 +1,134 @@
+          <label>{{ value[1].label }}</label>
+          <br />
+          <input :id="value[0]" type="text" :value="value[1].val" :label="value[1].label" @keyup="(payload) => converter(payload)" />
+          <br /><br />
+        </div>
+      </v-card-text>
+      <v-card-actions>
+        <v-btn class="messageBtn" block @click="setBaseValues(0)">
+          Reset
+        </v-btn>
+      </v-card-actions>
+    </v-card>
+  </v-dialog>
 </template>
 
 <script lang="ts" setup>
 import { useState } from '#/store/SimulatorStore/state'
 const SimulatorState = useState()
-import { setBaseValues } from '#/simulator/src/utils'
 import { onMounted, ref } from '@vue/runtime-core'
 
-const inputArr = ref([])
-inputArr.value = [
-    {
-        inputId: 'decimalInput',
-        val: '16',
-        label: 'Decimal value',
-    },
-    {
-        inputId: 'binaryInput',
-        val: '0b10000',
-        label: 'Binary value',
-    },
-    {
-        inputId: 'bcdInput',
-        val: '10110',
-        label: 'Binary-coded decimal vlaue',
-    },
-    {
-        inputId: 'octalInput',
-        val: '020',
-        label: 'Octal value',
-    },
-    {
-        inputId: 'hexInput',
-        val: '0x10',
-        label: 'Hexadecimal value',
-    },
-]
+const inputArr = ref(
+{
+'decimalInput' : {
+    val: '16',
+    label: 'Decimal value',
+  },
+  'binaryInput': {
+    val: '0b10000',
+    label: 'Binary value',
+  },
+  'bcdInput': {
+    val: '10110',
+    label: 'Binary-coded decimal value',
+  },
+  'octalInput': {
+    val: '020',
+    label: 'Octal value',
+  },
+  'hexInput': {
+    val: '0x10',
+    label: 'Hexadecimal value',
+  },
+})
 
 onMounted(() => {
-    SimulatorState.dialogBox.hex_bin_dec_converter_dialog = false
+  SimulatorState.dialogBox.hex_bin_dec_converter_dialog = false
 })
 
-function converter(feildChange) {
-    if (feildChange == 'decimalInput') decimalConvertor()
-    if (feildChange == 'binaryInput') binaryConvertor()
-    if (feildChange == 'bcdInput') bcdConvertor()
-    if (feildChange == 'octalInput') octalConvertor()
-    if (feildChange == 'hexInput') hexConvertor()
+function converter(e: KeyboardEvent) {
+  const target = <HTMLInputElement>e.target!;
+  switch(target.id) {
+  case 'decimalInput':
+    decimalConverter(target.value);
+    break;
+  case 'binaryInput':
+    binaryConverter(target.value);
+    break;
+  case 'bcdInput':
+    bcdConverter(target.value);
+    break;
+  case 'octalInput':
+    octalConverter(target.value);
+    break;
+  case 'hexInput':
+    hexConverter(target.value);
+    break;
+  }
 }
 
-function decimalConvertor() {
-    var x = parseInt($('#decimalInput').val(), 10)
-    setBaseValues(x)
+function setBaseValues(x: number) {
+  if (isNaN(x)) {
+    return;
+  }
+  inputArr.value.binaryInput.val = '0b' + x.toString(2);
+  inputArr.value.bcdInput.val = parseInt(x.toString(10), 16).toString(2);
```

Comment:
maybe we update the bcd function to always generate 4 binary digits 
wdyt ?

```
function convertToBCD(number) {
    let digits = number.toString().split('');
    let bcdOfDigits = digits.map(function(digit) {
        return parseInt(digit).toString(2).padStart(4, '0');
    });
    return bcdOfDigits.join('');
}
```
this would make sure to format the bcd correctly
like for `2` instead of `10` it would show `0010`